src/util.rs

```diff
@@ -183,20 +187,42 @@ where
     }
 }
 
-#[allow(dead_code)]
-pub fn convert_vec_f32(mut a: Vec<u8>, e: Endianness) -> Vec<f32> {
-    if e != Endianness::system() {
-        for c in a.chunks_mut(4) {
-            let (a, b) = c.split_at_mut(2);
-            mem::swap(&mut a[0], &mut b[1]);
-            mem::swap(&mut a[1], &mut b[0]);
+#[cfg(feature = "ndarray_volumes")]
+pub fn convert_bytes_and_cast_to<I, O>(
+    raw_data: Vec<u8>,
+    e: Endianness,
+    dim: &Vec<Ix>,
+    slope: O,
+    inter: O
+    ) -> Array<O, IxDyn>
+    where I: AsPrim + PodTransmutable,
+          O: AsPrim + Num
+{
+    let data: Vec<I> = convert_bytes_to(raw_data, e);
+    Array::from_shape_vec(IxDyn(&dim).f(), data)
+        .expect("Inconsistent raw data size")
+        .mapv(|v| raw_to_value::<I, O>(v, slope, inter))
+}
+
+#[cfg(feature = "ndarray_volumes")]
+pub fn convert_bytes_to<T: PodTransmutable>(
+    mut a: Vec<u8>,
+    e: Endianness
+) -> Vec<T> {
+    let nb_bytes = mem::size_of::<T>();
+    if e != Endianness::system() && nb_bytes > 1 {
+        // Swap endianness by block of nb_bytes
+        let split_at = nb_bytes / 2;
+        let split_at_m1 = split_at - 1;
+        for c in a.chunks_mut(nb_bytes) {
+            let (a, b) = c.split_at_mut(split_at);
+            for i in 0..split_at {
```

Comment:
Here's something more idiomatic:

```
for (l, r) in a.iter_mut().zip(b.iter_mut().rev()) {
    mem::swap(l, r);
}
```

No indexing operator, and no need for `split_at_m1`.

Reply:
Much better, thank you.